test/test_reserve_task_category.py

```diff
@@ -79,6 +79,15 @@ def test_task_category_to_sql_filter(self):
                 "reserve_task:project:202:category:x:1:y:2:z:3:user:1008:task:454"
             ], "filters, category must be as per keys passed and include negate, NOT IN clause"
 
+        # task category key exists, returns sql filter and its associated category_keys
+        project_id, exclude = "202", False
+        task_info = dict(name1="john's value", name2="john's baker's value")
+        expected_sql_filter = " AND ((task.info->>'name1' = 'john\'s value' AND task.info->>'name2' = 'john\'s baker\'s value')) "
```

Comment:
If changing the format for the escaping to double single quotes, the unit test would change accordingly.